.github/workflows/checks.yml

```diff
@@ -11,7 +11,7 @@ jobs:
   test:
     strategy:
       matrix:
-        python: [ "3.8", "3.9", "3.10", "3.11", "3.12", "3.13" ]
+        python: [ "3.9", "3.10", "3.11", "3.12", "3.13" ]
```

Comment:
I know there's another PR extant for dropping 3.8, but I had to remove 3.8 from the testing matrix here because 3.8 has some differences in the `typing` module that would make some of the new test assertions a pain to implement.